Sources/Customization/BatchEventProcessor.swift

```diff
@@ -0,0 +1,271 @@
+    // batch size (1 = no batching, 0 or negative = use default)
+    // attempt to send events in batches with batchSize number of events combined
+    var batchSize: Int
+    var maxQueueSize: Int
+    
+    public struct DefaultValues {
+        static public let batchSize = 10
+        static public let timeInterval: TimeInterval = 60  // secs
+        static public let maxQueueSize = 10000
+        static let maxFailureCount = 3
+    }
+    
+    // the max failure count.  there is no backoff timer.
+    
+    let eventHandler: OPTEventHandler
+    lazy var logger = OPTLoggerFactory.getLogger()
+    var backingStore: DataStoreType
+    var backingStoreName: String
+    
+    // for dispatching events
+    let dispatcher = DispatchQueue(label: "DefaultEventDispatcherQueue")
+    // using a datastore queue with a backing file
+    let dataStore: DataStoreQueueStackImpl<EventForDispatch>
+    // timer as a atomic property.
+    var timer: AtomicProperty<Timer> = AtomicProperty<Timer>()
+    
+    var observerProjectId: NSObjectProtocol?
+    var observerRevision: NSObjectProtocol?
+    
+    public init(eventHandler: OPTEventHandler = DefaultEventHandler(),
+                batchSize: Int = DefaultValues.batchSize,
+                timerInterval: TimeInterval = DefaultValues.timeInterval,
+                maxQueueSize: Int = DefaultValues.maxQueueSize,
+                backingStore: DataStoreType = .file,
+                dataStoreName: String = "OPTEventQueue") {
+        self.eventHandler = eventHandler
+        self.batchSize = batchSize > 0 ? batchSize : DefaultValues.batchSize
+        self.timerInterval = timerInterval >= 0 ? timerInterval : DefaultValues.timeInterval
+        self.maxQueueSize = maxQueueSize >= 100 ? maxQueueSize : DefaultValues.maxQueueSize
+        
+        self.backingStore = backingStore
+        self.backingStoreName = dataStoreName
+
+        switch backingStore {
+        case .file:
+            self.dataStore = DataStoreQueueStackImpl<EventForDispatch>(queueStackName: "OPTEventQueue",
+                                                                       dataStore: DataStoreFile<[Data]>(storeName: backingStoreName))
+        case .memory:
+            self.dataStore = DataStoreQueueStackImpl<EventForDispatch>(queueStackName: "OPTEventQueue",
+                                                                       dataStore: DataStoreMemory<[Data]>(storeName: backingStoreName))
+        case .userDefaults:
+            self.dataStore = DataStoreQueueStackImpl<EventForDispatch>(queueStackName: "OPTEventQueue",
+                                                                       dataStore: DataStoreUserDefaults())
+        }
+        
+        if self.maxQueueSize < self.batchSize {
+            self.logger.e(.eventDispatcherConfigError("batchSize cannot be bigger than maxQueueSize"))
+            self.maxQueueSize = self.batchSize
+        }
+        
+        addProjectChangeNotificationObservers()
```

Comment:
I'd like to remove this and use the notification center as a property that is injected.